libethereum/EthereumPeer.cpp

```diff
@@ -106,22 +103,22 @@ void EthereumPeer::requestStatus()
 
 void EthereumPeer::requestHashes()
 {
-	if (m_asking == Asking::Blocks)
-		return;
+	assert(m_asking == Asking::Nothing);
```

Comment:
this is /very/ harsh; why bother crashing the app in the case, rather than attempting to recover?